File: lab14/app/src/main/java/com/example/lab14/MainActivity.java
package com.example.lab14;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private Button checkButton;
    private ImageView offlineImage, onlineImage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        checkButton = findViewById(R.id.checkButtonId);
        System.out.println("images init");
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    public void onCheckButtonClick(View view) {
        ConnectivityManager connectivityManager = (ConnectivityManager)
                getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        offlineImage = (ImageView) findViewById(R.id.offlineImage);
        onlineImage = (ImageView) findViewById(R.id.onlineImage);

        if (networkInfo != null && networkInfo.isConnected()) {
            offlineImage.setVisibility(View.INVISIBLE);
            onlineImage.setVisibility(View.VISIBLE);
        } else {
            offlineImage.setVisibility(View.VISIBLE);
            onlineImage.setVisibility(View.INVISIBLE);
        }
    }
}

File: lab_1/app/src/main/java/com/example/app_1/MainActivity.kt
package com.example.app_1

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val startBtn: ImageButton = findViewById(R.id.helloButton)
        val text: TextView = findViewById(R.id.helloText)
        val image: ImageView = findViewById(R.id.helloImage)

        startBtn.setOnClickListener(View.OnClickListener {
            startBtn.visibility = View.INVISIBLE
            text.visibility = View.VISIBLE
            image.visibility = View.VISIBLE
            startBtn.isClickable = false
        })

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
}